Use StageFromReader instead of creation of temporary file

evgmik opened the pull request:
This fixes #58. Old way: to stage from stdin a temporary file was created
but this file was not removed after use. Which polluted temporary
directories with such files.

New way: directly stage from stdin stream with StageFromReader.
Oddly, this function was already existing but was not used.

evgmik committed:
Use StageFromReader instead of creation of temporary file

This fixes #58. Old way: to stage from stdin a temporary file was created
but this file was not removed after use. Which polluted temporary
directories with such files.

New way: directly stage from stdin stream with StageFromReader.
Oddly, this function was already existing but was not used.

sahib reviewed (approved):
Approved, but I found a mistake of myself:

https://github.com/sahib/brig/blob/3c880455d299a46e6110c5032b0c2d367225831c/client/fs_cmds.go#L150

There should be a fd.Close() before the Stage() to not leak a file descriptor and make sure all content is flushed to the file.

evgmik committed:
Fix: Closing temporary file before passing it for staging

StageFromReader creates a temporary file from the stream, but it was not
closed after use. This was an opportunity for memory leaks or saturation
of allowed number for file handles.
